Document predicate pushdown support for string-type columns in SQL Se… (#16366)
trinodb/trino: docs/src/main/sphinx/connector/sqlserver.rst

.. include:: join-pushdown-enabled-true.fragment

Predicate pushdown support
""""""""""""""""""""""""""

> The connector supports pushdown of any predicates on ``VARCHAR`` and

Reviewer: `any` isn't true as showcased by all the caveats that follow

```suggestion
The connector supports pushdown of predicates on ``VARCHAR`` and
```

.. include:: join-pushdown-enabled-true.fragment

Predicate pushdown support
""""""""""""""""""""""""""

The connector supports pushdown of predicates on ``VARCHAR`` and ``NVARCHAR``
columns if the underlying columns in SQL Server use a case-sensitive `collation
<https://learn.microsoft.com/en-us/sql/relational-databases/collations/collation-and-unicode-support?view=sql-server-ver16>`_.
If the column uses a case-insensitive collation, then no operators are pushed

> down to preserve correct results.

Reviewer: Should this statement come AFTER the list? Otherwise this feels confusing.

.. include:: join-pushdown-enabled-true.fragment

Predicate pushdown support
""""""""""""""""""""""""""

The connector supports pushdown of predicates on ``VARCHAR`` and ``NVARCHAR``
columns if the underlying columns in SQL Server use a case-sensitive `collation
<https://learn.microsoft.com/en-us/sql/relational-databases/collations/collation-and-unicode-support?view=sql-server-ver16>`_.

The following operators are pushed down:

- ``=``
- ``<>``
- ``IN``
- ``NOT IN``

> Operators are not pushed down and correct results are not preserved for columns
> using a case-insensitive collation.

Reviewer: ```suggestion
No operators are pushed down for columns using
a case-insensitive collation to ensure correct results.
```

Reviewer 2: I feel like "Operators are not pushed down" reads better than "No operators are pushed down" without changing the meaning.

> Operators are not pushed down and correct results are not preserved for columns
> using a case-insensitive collation.

Reviewer: ```suggestion
Operators are not pushed down for columns using a case-insensitive collation
to preserve correct results.
```

the current sentence reads the opposite (i.e. results are incorrect)

Author: It's the "to preserve correct results" that needs to be changed to sound less awkward. Since I'm having trouble improving the flow of the sentence without changing the meaning, please help me understand this process better.

For columns using a case-insensitive collation, operators are not pushed down, which preserves correct results. Is that right or wrong?

Reviewer: > For columns using a case-insensitive collation, operators are not pushed down, which preserves correct results.

Yes, this is the behaviour and intention.

.. include:: join-pushdown-enabled-true.fragment

Predicate pushdown support
""""""""""""""""""""""""""

The connector supports pushdown of any predicates on ``VARCHAR`` and
``NVARCHAR`` columns. The following operators are supported:

> - EQUAL

Reviewer 3: I dont think we use that in others .. it probably more like =, !=  .. check with others

- NOT EQUAL
- IN
- NOT IN

> .. code-block:: sql

Reviewer 3: I dont think we need an example

.. example

> To enable pushdown for case-sensitive columns, change the collation setting to a

Reviewer 3: I have no idea what that means. What setting do you change from what to what?

Reviewer 4: In SqlServer every textual column have a Collation.
Collation it's kind of set of rules how to treat text symbols in this column during different operations.
For example collation Latin1_General_CI_AS - means for this column we translate bytes to symbols using Latin byte to symbol translation and CI -  mean Case Insensitivity and other rules.
So our code is working only if collation is Case Sensitive for column.

Reviewer 4: Case sensitive collation means that in Collation name there is part like _CS_ or _BIN_ or _BIN2_
